src/cascadia/TerminalApp/App.cpp

```diff
@@ -713,14 +714,32 @@ namespace winrt::TerminalApp::implementation
     void App::_UpdateTitle(std::shared_ptr<Tab> tab)
     {
         auto newTabTitle = tab->GetFocusedTitle();
+        const auto lastFocusedProfileOpt = tab->GetFocusedProfile();
+        const auto lastFocusedProfile = lastFocusedProfileOpt.value();
+        const auto* const matchingProfile = _settings->FindProfile(lastFocusedProfile);
+
+        auto tabTitle = matchingProfile->GetTabTitle();
 
         // TODO #608: If the settings don't want the terminal's text in the

```

Comment:
Do we still need this TODO? Maybe replace it with the data binding one or just remove it entirely.